### DISCUSSION TITLE
StokesFlow.run() show "A or b contains inf/nan values"

### DISCUSSION THREAD

**Discussion by MJL-JLU** [PMEAL/OpenPNM#2779] [Q&A]
Dear professor
      I generated a batch of model data using avizo and porespy,  but I got this error when running the _flow_run_ function of the Predicting absolute permeability case, I tried several models, only one worked, how should I fix it.

**Comment by jgostick** [comment]
It often happens if your pores are too big and overlap.

**Comment by MJL-JLU** [comment]
> It often happens if your pores are too big and overlap.

Thank you very much for your reply, Professor, is there any way to solve this problem, can this only be determined by the characteristics of the model itself?

**Reply by gxw-ll** [reply]
Hi, MJL, Have you solved the problem?

**Reply by MJL-JLU** [reply]
I 'm sorry I 've only seen it now, but I later discovered that this happened because of the presence of nan in phase [ ' throat.hydraulic _ conductance ' ], and I found them and assigned them to 0

**Reply by gxw-ll** [reply]
Thank you very much for your answer！

**Comment by jgostick** [comment]
start inspecting your pores sizes via graphs, look into the specific values to find the ones that are a probably.  It takes a bit of investigation. If you are using a generated network, then you can increase the spacing too.

**Comment by likeduck** [comment]
There are many reasons for this. If you extract PNM based on porous media, this problem generally does not occur. But if it is a randomly generated PNM, this situation can easily occur due to unreasonable parameters. For example, like Prof.'s comment, the pore diameter in your model is too large causing pore overlap, which "could" lead to uncertainty in the throat, as we know that pressure drop in PNM occurs in the throat. Another situation I encountered was that the newly created throat was not assigned a value when suturing two PNMs. In short, the cause of this problem is difficult to check. You can change the model and try again after success.
